bentoml/xgboost.py

```diff
@@ -30,34 +35,60 @@
         "pd", globals(), "pandas", exc_msg="Install pandas with `pip install pandas`"
     )
 
+LOAD_RETURNS_DOCS = """\
+    Returns:
+        an instance of `xgboost.core.Booster` from BentoML modelstore.
+"""
 
+
+@init_docstrings(LOAD_INIT_DOCS)
+@returns_docstrings(LOAD_RETURNS_DOCS)
+def load(
+    name: str,
+    infer_params: t.Dict[str, t.Union[str, int]] = None,
+    nthread: int = -1,
+) -> "xgb.core.Booster":
+    """
+    infer_params (`t.Dict[str, t.Union[str, int]]`):
+        Params for booster initialization
+    nthread (`int`, default to -1):
+        Number of thread will be used for this booster.
+         Default to -1, which will use XgBoost internal threading
+         strategy.
+    """
+    model_info = _stores.get(name)
+    if infer_params is None:
+        infer_params = model_info.options
+    if "nthread" not in infer_params:
+        infer_params["nthread"] = nthread
+    return xgb.core.Booster(
+        params=infer_params,
+        model_file=os.path.join(model_info.path, f"{SAVE_NAMESPACE}{JSON_EXT}"),
+    )
+
+
+@init_docstrings(SAVE_INIT_DOCS)
+@returns_docstrings(SAVE_RETURNS_DOCS)
 def save(
     name: str,
     model: "xgb.core.Booster",
     *,
     infer_params: t.Dict[str, t.Union[str, int]] = None,
     metadata: t.Optional[GenericDictType] = None,
 ) -> str:
-    cur_frame = sys._getframe().f_back.f_code.co_filename
-    _instance = _XgBoostModel(model, metadata=metadata)
-    save_options = dict(infer_params=infer_params)
+    """
+    model (`xgboost.core.Booster`):
+        instance of model to be saved
+    """
     with _stores.register(
-        name, module=_instance.__module__, save_options=save_options, stacks=cur_frame
+        name, module=__name__, options=infer_params, metadata=metadata
     ) as ctx:
-        _version = ctx.version
-        _instance.save(ctx.path)
-    return f"{name}:{_version}"
+        model.save_model(os.path.join(ctx.path, f"{SAVE_NAMESPACE}{JSON_EXT}"))
+    return f"{name}:{ctx.version}"
 
 
-def load(name: str, **load_kwargs) -> "xgb.core.Booster":
-    model_info = _stores.get(name)
```

Comment:
this actually looks right to me. And since `load` and `load_runner` are user-facing, we should put the detailed arguments list and docstring here